tests/Service/PromotionsService/AbstractPromotionServiceTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\PromotionsService;
+
+use BBC\ProgrammesPagesService\Domain\Entity\Brand;
+use BBC\ProgrammesPagesService\Domain\Entity\Promotion;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use BBC\ProgrammesPagesService\Service\PromotionsService;
+use Tests\BBC\ProgrammesPagesService\AbstractServiceTest;
+
+abstract class AbstractPromotionServiceTest extends AbstractServiceTest
+{
+    public function setUp()
+    {
+        $this->setUpCache();
+        $this->setUpRepo('PromotionRepository');
+        // override getDomainModel() in mapper with getDomainModelFromDbData()
+        $this->setUpMapper('PromotionMapper', 'getDomainModelFromDbData');
+    }
+
+    /**
+     * @param mixed[] $dbPromotion
+     */
+    protected function getDomainModelFromDbData(array $dbPromotion): Promotion
```

Comment:
I'm suspicious of this method. It doesn't seem to be something that should be needed in a test.